vochain/state.go

```diff
@@ -424,50 +554,48 @@ func (v *State) AddVote(vote *models.Vote) error {
 	return nil
 }
 
-// voteID = byte( processID+nullifier )
+// NOTE(Edu): Changed this from byte(processID+nullifier) to
+// hash(processID+nullifier) to allow using it as a key in Arbo tree.
+// voteID = hash(processID+nullifier)
 func (v *State) voteID(pid, nullifier []byte) ([]byte, error) {
 	if len(pid) != types.ProcessIDsize {
 		return nil, fmt.Errorf("wrong processID size %d", len(pid))
 	}
 	if len(nullifier) != types.VoteNullifierSize {
 		return nil, fmt.Errorf("wrong nullifier size %d", len(nullifier))
 	}
-	vid := bytes.Buffer{}
+	vid := sha256.New()
 	vid.Write(pid)
 	vid.Write(nullifier)
-	return vid.Bytes(), nil
+	return vid.Sum(nil), nil
 }
 
 // Envelope returns the hash of a stored vote if exists.
 func (v *State) Envelope(processID, nullifier []byte, isQuery bool) (_ []byte, err error) {
-	// TODO(mvdan): remove the recover once
-	// https://github.com/tendermint/iavl/issues/212 is fixed
```

Comment:
they still haven't fixed this :see_no_evil: good that we're moving away from IAVL I think.